clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -1082,15 +1154,22 @@
     /* ApiVersion api */
     public static final Schema API_VERSIONS_REQUEST_V0 = new Schema();
 
+    /* v1 request is the same as v0. Throttle time has been added to response */
+    public static final Schema API_VERSIONS_REQUEST_V1 = API_VERSIONS_REQUEST_V0;
+
     public static final Schema API_VERSIONS_V0 = new Schema(new Field("api_key", INT16, "API key."),
                                                            new Field("min_version", INT16, "Minimum supported version."),
                                                            new Field("max_version", INT16, "Maximum supported version."));
 
     public static final Schema API_VERSIONS_RESPONSE_V0 = new Schema(new Field("error_code", INT16, "Error code."),
                                                                     new Field("api_versions", new ArrayOf(API_VERSIONS_V0), "API versions supported by the broker."));
+    public static final Schema API_VERSIONS_RESPONSE_V1 = new Schema(
+            new Field("error_code", INT16, "Error code."),
+            new Field("api_versions", new ArrayOf(API_VERSIONS_V0), "API versions supported by the broker."),
+            newThrottleTimeField());
```

Comment:
Ok, could we add a comment that error_code has to be the first field in ApiResponse?